src/vdaf/prio3/l1boundsum.rs

```diff
@@ -0,0 +1,67 @@
+//! Implementation of the Prio3L1BoundSum VDAF defined in [1].
+//!
+//! [1]: https://martinthomson.github.io/prio-l1-bound-sum/draft-thomson-ppm-l1-bound-sum.html
+
+use crate::{
+    field::Field128,
+    flp::{
+        gadgets::{Mul, ParallelSum},
+        types::l1boundsum::L1BoundSum,
+    },
+    vdaf::{prio3::Prio3, xof::XofTurboShake128, VdafError},
+};
+
+/// A VDAF which sums over vectors, checking that each element of each vector is within some range
+/// and that the L1 norm of each vector matches a claimed weight encoded into the input, which
+/// itself must be within the same range as the vector elements, effectively bounding the L1 norm
+/// to 2^bits - 1.
+pub type Prio3L1BoundSum =
+    Prio3<L1BoundSum<Field128, ParallelSum<Field128, Mul<Field128>>>, XofTurboShake128, 32>;
+
+impl Prio3L1BoundSum {
+    /// Construct an instance of Prio3L1BoundSum with the given number of aggregators. `bits` defines
+    /// the bit width of each summand of the measurement; `len` defines the length of the
+    /// measurement vector.
+    pub fn new_l1_bound_sum(
+        num_aggregators: u8,
+        bits: usize,
+        len: usize,
+        chunk_length: usize,
+    ) -> Result<Self, VdafError> {
+        Prio3::new(
+            num_aggregators,
+            1,
+            // TODO: use real codepoint once assigned
+            0xFFFF0000,
```

Comment:
The private use codepoint might be reason enough to put this behind the `experimental` feature for now.